synapse/push/push_tools.py

```diff
@@ -49,7 +50,33 @@ async def get_badge_count(store: DataStore, user_id: str, group_by_room: bool) -
 async def get_context_for_event(
     storage: StorageControllers, ev: EventBase, user_id: str
 ) -> Dict[str, str]:
-    ctx = {}
+    ctx: Dict[str, str] = {}
+
+    if ev.internal_metadata.outlier:
+        # We don't have state for outliers, so we can't compute the context
+        # except for invites with `invite_room_state` set..
```

Comment:
It would be nice to call out "out-of-band membership" specifically here so people can cross-reference to https://github.com/matrix-org/synapse/blob/develop/docs/development/room-dag-concepts.md#out-of-band-membership-events more easily to get more context